miner/worker.go

```diff
@@ -383,10 +410,23 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 		case <-w.startCh:
 			clearPending(w.chain.CurrentBlock().NumberU64())
 			timestamp = time.Now().Unix()
+			//reset txpoolsnapshot and interrupt current pre-mining
+			select {
+			case w.resetPoolSnapshot <- struct{}{}:
+			default:
+			}
+			select {
+			case w.preMiningInterruptOnMining <- struct{}{}:
+			default:
+			}
 			commit(true, commitInterruptNewHead)
 
 		case head := <-w.chainHeadCh:
 			if !w.isRunning() {
+				select {
+				case w.txpoolChainHeadCh <- struct{}{}:
```

Comment:
worker is not running, when downloader is started, it is for quick block catch up mode.
when downloader is done, the worker will be restarted, but downloader could be started again to downloader more header, block, receipt...
When downloader is running, it is better not to do TxPoolPrefetch.

Actually, TxPoolPrefetch should be executed only when the node has synced to the latest block